File: NetDeck/SmallCardCell.m
//
//  SmallCardCell.m
//  Net Deck
//
//  Created by Gereon Steffens on 27.01.14.
//  Copyright © 2016 Gereon Steffens. All rights reserved.
//

#import "SmallCardCell.h"
#import "CGRectUtils.h"

@implementation SmallCardCell

-(void) awakeFromNib
{
    self.name.font = [UIFont md_mediumSystemFontOfSize:17];
    
    NSInteger diameter = 8;
    self.mwlMarker.frame = CGRectSetSize(self.mwlMarker.frame, diameter, diameter);
    self.mwlMarker.layer.cornerRadius = diameter/2;
    self.mwlMarker.layer.backgroundColor = [UIColor whiteColor].CGColor;
    self.mwlMarker.layer.borderWidth = 1;
    self.mwlMarker.layer.borderColor = [UIColor blackColor].CGColor;
}

-(void) setCardCounter:(CardCounter *)cc
{
    [super setCardCounter:cc];
    Card* card = cc.card;
    
    self.copiesStepper.hidden = card.type == NRCardTypeIdentity;
    self.identityButton.hidden = card.type != NRCardTypeIdentity;
    self.factionLabel.hidden = card.type == NRCardTypeIdentity;
    
    if (cc == nil)
    {
        [self.identityButton setTitle:l10n(@"Choose Identity") forState:UIControlStateNormal];
    }
    else
    {
        [self.identityButton setTitle:l10n(@"Switch Identity") forState:UIControlStateNormal];
    }
    
    self.copiesStepper.maximumValue = self.deck.isDraft ? 100 : cc.card.maxPerDeck;
    self.copiesStepper.value = cc.count;
    
    if (card.type == NRCardTypeIdentity)
    {
        self.name.text = card.name;
    }
    else if (card.unique)
    {
        self.name.text = [NSString stringWithFormat:@"%lu× %@ •", (unsigned long)cc.count, card.name];
    }
    else
    {
        self.name.text = [NSString stringWithFormat:@"%lu× %@", (unsigned long)cc.count, card.name];
    }
    
    self.name.textColor = [UIColor blackColor];
    if (!self.deck.isDraft && card.owned < cc.count)
    {
        self.name.textColor = [UIColor redColor];
    }
    
    NSUInteger influence = 0;
    if (card.type == NRCardTypeAgenda)
    {
        influence = card.agendaPoints * cc.count;
    }
    else
    {
        if (self.deck)
        {
            influence = [self.deck influenceFor:cc];
        }
        else
        {
            influence = card.influence * cc.count;
        }
    }
    
    if (influence > 0)
    {
        self.influenceLabel.textColor = card.factionColor;
        self.influenceLabel.text = [NSString stringWithFormat:@"%lu", (unsigned long)influence];
    }
    else
    {
        self.influenceLabel.text = @"";
    }
    
    self.mwlMarker.hidden = !card.isMostWanted;
    self.factionLabel.text = card.factionStr;
}

@end

File: NetDeck/SmallCardCell.swift
//
//  SmallCardCell.swift
//  NetDeck
//
//  Created by Gereon Steffens on 31.12.16.
//  Copyright © 2021 Gereon Steffens. All rights reserved.
//

import UIKit

class SmallCardCell: CardCell {
    
    @IBOutlet weak var factionLabel: UILabel!
    @IBOutlet weak var mwlMarker: UIView!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        self.name.font = UIFont.monospacedDigitSystemFont(ofSize: 17, weight: UIFont.Weight.regular)
        
        let diameter: CGFloat = 8
        let frame = CGRect(origin: self.mwlMarker.frame.origin, size: CGSize(width: diameter, height: diameter))
        self.mwlMarker.frame = frame
        self.mwlMarker.layer.cornerRadius = diameter / 2.0
        self.mwlMarker.layer.backgroundColor = UIColor.systemBackground.cgColor
        self.mwlMarker.layer.borderColor = UIColor.label.cgColor
        self.mwlMarker.layer.borderWidth = 1
        
        self.prepareForReuse()
    }
    
    override func prepareForReuse() {
        super.prepareForReuse()
        
        self.mwlMarker.isHidden = true
        self.factionLabel.text = ""
    }
    
    
    override func setCardCounter(_ cardCounter: CardCounter?) {
        super.setCardCounter(cardCounter)
        
        guard let cc = cardCounter else {
            return
        }
        
        let card = cc.card
        
        self.copiesStepper.isHidden = card.type == .identity
        self.identityButton.isHidden = card.type != .identity
        self.factionLabel.isHidden = card.type == .identity

        self.copiesStepper.maximumValue = Double(self.deck.isDraft ? 100 : cc.card.maxPerDeck)
        self.copiesStepper.value = Double(cc.count)
        
        self.name.text = cc.displayName(self.deck.mwl)
        
        self.name.textColor = .label
        if !self.deck.isDraft && (cc.count > card.owned || card.isRotated) {
            self.name.textColor = .systemRed
        }
        if card.banned(self.deck.mwl) {
            self.name.textColor = .systemRed
        }
        
        let influence = self.deck.influenceFor(cc)
        
        if influence > 0 {
            self.influenceLabel.textColor = card.factionColor
            self.influenceLabel.text = "\(influence)"
        } else {
            self.influenceLabel.text = ""
        }
        
        self.factionLabel.text = card.factionStr
        
        let penalty = card.mwlPenalty(self.deck.mwl)
        self.mwlMarker.isHidden = penalty == 0
    }
}
